pydeseq2/utils.py

```diff
@@ -170,14 +170,33 @@ def build_design_matrix(
         A DataFrame with experiment design information (to split cohorts).
         Indexed by sample barcodes.
     """
+
     design_matrix = pd.get_dummies(clinical_df[design])
+
+    if design_matrix.shape[-1] == 1:
+        raise ValueError(
+            f"The design factor takes only one value: "
+            f"{design_matrix.columns.values.tolist()}, but two are necessary."
+        )
+    elif design_matrix.shape[-1] > 2:
+        raise ValueError(
+            f"The design factor takes more than two values: "
+            f"{design_matrix.columns.values.tolist()}, but should have exactly two."
+        )
     # Sort columns alphabetically
     design_matrix = design_matrix.reindex(
         sorted(design_matrix.columns, reverse=True), axis=1
     )
-    if ref is not None:  # Put reference level first
-        ref_level = design_matrix.pop(ref)
-        design_matrix.insert(0, ref, ref_level)
+    if ref is not None:  # Put reference level last
+        try:
+            ref_level = design_matrix.pop(ref)
+        except KeyError as e:
+            print(
+                "The reference level must correspond to"
+                " one of the design factor values."
+            )
+            raise e
```

Comment:
I am more used to 

```python
except KeyError as e:
            raise e( "The reference level must correspond to"
            +  " one of the design factor values.")
```
But I don't know if one or the other is best. What do you think ?

Reply:
Indeed

Reply:
you are right, 
The syntax should be 
```python
try:
   pass
except KeyError as e:
   raise KeyError('mon text') from e
``` 
Or create a custom exception. I wonder if the print in an except goes to sys.stdout or sys.stderr